src/wrapper/strings/ffi_str.rs

```diff
@@ -28,7 +29,7 @@ impl ::std::ops::Deref for JNIString {
     type Target = JNIStr;
 
     fn deref(&self) -> &Self::Target {
-        unsafe { ::std::mem::transmute(self.internal.as_bytes_with_nul()) }
+        unsafe { &*(self.internal.as_bytes_with_nul() as *const [u8] as *const ffi_str::JNIStr) }
```

Comment:
This double "as" confuses me a little :smile:

Reply:
The first `as` converts the shared reference into the const pointer, the second `as` converts the type of the pointer. Both at the same time is too implicit and may lead to an error, I believe.